pending/json-api.pod

```diff
@@ -0,0 +1,290 @@
+                },
+            },
+            {
+                type => 'gift',
+                id => 'g998383',
+                attributes => {
+                    name => 'Verg articulated figure'
+                },
+            }
+        ]
+    }
+
+Now, I know what you are thinking: "HOLLY DECKS THE HALLS, that's verbose!".
+And yes, yes it is. JSON:API doesn't aspire to be terse or pretty, but rather to
+be consistent and straightforward to parse. 
+
+I won't go into the nitty gritties, but the main things to know about 
+a JSON:API serialized object is that you have a wee bit of meta information
+
+        jsonapi => { version => '1.0 '},
+        links => {
+            self => '/child/cromedome',
+        },
+
+and then you always have the object's type and id,
+
+        data => {
+            id => 'child-83626812-b',
+            type => 'child',
+
+you B<can> have attributes of that object, 
+
+            attributes => {
+                naughtoscale => 8,
+                name => 'cromedome',
+            }, 
+
+and you B<can> have a list of related objects.
+
+            relationships=> {
+                gifts => {
+                   links => {
+                        self => '/child/cromedone/wishlist',
+                   },
+                   data => [
+                        { type => 'gift', id => 'g829137' },
+                        { type => 'gift', id => 'g998383' },
+                   ] 
+                }
+            }
+
+It's also possible to have those related objects fleshed out. In that case,
+the information in the C<relationships> hash still stays the C<type>/C<id>
+pair, but the full object would be in an C<included> field.
+
+        included => [
+            {
+                type => 'gift',
+                id => 'g829137',
+                attributes => {
+                    name => 'Cessna Skyhawk'
+                },
+            },
+            {
+                type => 'gift',
+                id => 'g998383',
+                attributes => {
+                    name => 'Verg articulated figure'
+                },
+            }
+        ]
+    }
+
+
+
+As for all those sorting / filtering / including tweaks, the JSON:API specs
+prescribe how to convey them via the endpoint query path. Requesting a subset
+of the attributes is 
+done via C<?fields=name,naughtoscale>. Adding related objects to
+the payload is done via C<?include=gifts>. Pagination is done via
+C<?page[number]=X>  or perhaps C<?page[offset]=Y>.  Is this way of doing thing revolutionary? Not at all.
+Chances are you are already using very keywords. But the point is that it's
+setting a very well-defined mold. No need to ever agonize about the details of
+any of those parameters; just follow the recipe.
+
+=head2 Cue in Dancer2::Plugin::JsonApi
+
+Now, how does that apply to the dance floor? Glad you asked, for there is a
+(very) new plugin in town, L<Dancer2::Plugin::ApiJson> to help. 
```

Comment:
JsonApi, not ApiJson, right?

Reply:
That's what I get for hiring Yoda as a copy editor... >.>